tsl/test/expected/hypercore_dump_restore.out

```diff
@@ -0,0 +1,310 @@
+ _timescaledb_internal._hyper_1_2_chunk
+ _timescaledb_internal._hyper_1_3_chunk
+ _timescaledb_internal._hyper_1_4_chunk
+ _timescaledb_internal._hyper_1_5_chunk
+ _timescaledb_internal._hyper_1_6_chunk
+(6 rows)
+
+reindex table hyperdump;
+explain (costs off)
+select * from hyperdump where device = 2 and time < '2022-06-03';
+                                                 QUERY PLAN                                                 
+------------------------------------------------------------------------------------------------------------
+ Append
+   ->  Index Scan using _hyper_1_1_chunk_time_device_idx on _hyper_1_1_chunk
+         Index Cond: ((device = 2) AND ("time" < 'Fri Jun 03 00:00:00 2022 PDT'::timestamp with time zone))
+   ->  Index Scan using _hyper_1_2_chunk_time_device_idx on _hyper_1_2_chunk
+         Index Cond: ((device = 2) AND ("time" < 'Fri Jun 03 00:00:00 2022 PDT'::timestamp with time zone))
+(5 rows)
+
+set timescaledb.hypercore_copy_to_behavior='all_data';
+select chunk from show_chunks('hyperdump') chunk offset 2 limit 1 \gset
+\d+ :chunk
+                                                                   Table "_timescaledb_internal._hyper_1_3_chunk"
+ Column |           Type           | Collation | Nullable |                                     Default                                     | Storage  | Stats target | Description 
+--------+--------------------------+-----------+----------+---------------------------------------------------------------------------------+----------+--------------+-------------
+ time   | timestamp with time zone |           | not null |                                                                                 | plain    |              | 
+ device | integer                  |           |          |                                                                                 | plain    |              | 
+ tempc  | double precision         |           |          |                                                                                 | plain    |              | 
+ tempf  | double precision         |           |          | generated always as (tempc * 2::double precision + 30::double precision) stored | plain    |              | 
+ status | text                     |           |          | 'idle'::text                                                                    | extended |              | 
+Indexes:
+    "_hyper_1_3_chunk_time_device_idx" btree (device, "time" DESC)
+Check constraints:
+    "constraint_3" CHECK ("time" >= 'Wed Jun 08 17:00:00 2022 PDT'::timestamp with time zone AND "time" < 'Wed Jun 15 17:00:00 2022 PDT'::timestamp with time zone)
+Inherits: hyperdump
+
+SELECT format('%I.%I', c2.schema_name, c2.table_name)::regclass AS cchunk
+FROM _timescaledb_catalog.chunk c1
+INNER JOIN _timescaledb_catalog.chunk c2
+ON (c1.compressed_chunk_id = c2.id) offset 2 limit 1 \gset
+-- This should not return any data when in this COPY mode.
+copy :cchunk to stdout;
```

Comment:
Do you have tests a normal copy using all combinations {hypertable, chunk} x {all_data, no_compressed_data}?

Reply:
The issue here is that the other combos return data and I am not sure how to handle that. Most of them are tested automatically via the pgdump tests where the output is dumped to files and then compared with dumps from "regular" tables.

This copy command was easy to test explicitly because it should not return any data. It is also tested by pgdump, fwiw.

Reply:
> The issue here is that the other combos return data and I am not sure how to handle that.

Ok. IC.

Have you done a manual test and made sure that it looks sane?

> Most of them are tested automatically via the pgdump tests where the output is dumped to files and then compared with dumps from "regular" tables.
> 
> This copy command was easy to test explicitly because it should not return any data. It is also tested by pgdump, fwiw.

Ok. So it seems we have good coverage for the testing then.

My first thinking is that this might be more suitable as a TAP test, but maybe we should take that as a separate pull request.